gel/_internal/_save.py

```diff
@@ -1671,6 +1710,29 @@ def _traverse(obj: GelModel) -> None:
                         # see the above comment.
                         multi_prop_commit_recursive(linked)
 
+            if (
+                self.refetch
+                and obj in self.new_object_ids
+                and (new_obj_id := self.new_object_ids.get(obj))
+            ):
+                new_obj = self.new_objects.get(new_obj_id)
```

Comment:
What's going on here? Why does new_objects need a different path for updating the objects with refetched data than the old objects?